seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/EngineType.java

```diff
@@ -17,18 +17,51 @@
 
 package org.apache.seatunnel.config;
 
+import org.apache.seatunnel.flink.BaseFlinkSink;
+import org.apache.seatunnel.flink.BaseFlinkSource;
+import org.apache.seatunnel.flink.BaseFlinkTransform;
+import org.apache.seatunnel.spark.BaseSparkSink;
+import org.apache.seatunnel.spark.BaseSparkSource;
+import org.apache.seatunnel.spark.BaseSparkTransform;
+
+import java.util.HashMap;
+import java.util.Map;
+
 public enum EngineType {
-    SPARK("spark"),
-    FLINK("flink"),
+
+    SPARK("spark", new HashMap<PluginType, Class<?>>() {
+        {
+            put(PluginType.SOURCE, BaseSparkSource.class);
+            put(PluginType.TRANSFORM, BaseSparkTransform.class);
+            put(PluginType.SINK, BaseSparkSink.class);
+        }
+    }),
+
+    FLINK("flink", new HashMap<PluginType, Class<?>>() {
+        {
+            put(PluginType.SOURCE, BaseFlinkSource.class);
+            put(PluginType.TRANSFORM, BaseFlinkTransform.class);
+            put(PluginType.SINK, BaseFlinkSink.class);
+        }
+    }),
```

Comment:
Only static class initializers should be used

Reply:
Ok, add a PluginFactory class to do the create plugins work.